src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -708,7 +708,7 @@ FixedwingPositionControl::set_control_mode_current(const hrt_abstime &now)
 
 		} else if (_pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
 
-			if (!_vehicle_status.in_transition_mode) {
+			if (!(_vehicle_status.is_vtol && _param_nav_force_vtol.get() > 0)) {
 
```

Comment:
@KonradRudin I don't understand your point. 

> Now you are dependent, that somewhere exxternally the transition command will be executed

We have always been dependent on someone externally triggering the transition. It's navigator who publishes the transition command the the vtol controller executes. Not that I am a fan of that, but that's how it currently is.

> Why did you change that? I think it is because you want to enter the FW_POSCTRL_MODE_APPROACH_TRANSITON already before the transition happens, but is this really necessary? 

Exactly, I want the controller to be aware of that, because then it can specifically execute. E.g. it will not try to loiter even if there is a large altitude error. Only handling the transition is not enough, because you might have already started loitering. And honestly, I don't want to extend the logic in navigator.